config/job.js

```diff
@@ -77,34 +77,61 @@ const SCHEMA_ENVIRONMENT = Joi.object()
     + 'and underscore (cannot start with digit)'
     });
 const SCHEMA_JOBNAME = Joi.string().max(100).regex(Regex.JOB_NAME);
+// Step can be in the following formats:
+// npm install
+// { init: npm install }
+// { init: { command: npm install } }
 const SCHEMA_STEP_STRING = Joi.string();
+const SCHEMA_STEP_SUBOBJECT_BASE = Joi.object()
+    .keys({
+        command: SCHEMA_STEP_STRING.required()
+    });
+const SCHEMA_STEP_COMMAND = Joi.alternatives().try(SCHEMA_STEP_STRING, SCHEMA_STEP_SUBOBJECT_BASE);
 const SCHEMA_STEP_OBJECT = Joi.object()
     // Steps can only be named with A-Z,a-z,0-9,-,_
     // Steps only contain strings (the command to execute)
-    .pattern(Regex.STEP_NAME, Joi.string())
+    .pattern(Regex.STEP_NAME, SCHEMA_STEP_COMMAND)
     // All others are marked as invalid
     .unknown(false)
-    // And there can be only one command per step
+    // And there can be only one command per step or an object with command
     .length(1)
     // Add documentation
     .messages({ 'object.unknown': '{{#label}} only supports the following '
     + 'characters A-Z,a-z,0-9,-,_' });
-
-const SCHEMA_DESCRIPTION = Joi.string().max(100).optional();
-const SCHEMA_IMAGE = Joi.string().regex(Regex.IMAGE_NAME);
-const SCHEMA_ORDER = Joi.array()
-    .items(SCHEMA_STEP_STRING.regex(Regex.STEP_NAME))
-    .min(0);
-const SCHEMA_SETTINGS = Joi.object().optional();
 const SCHEMA_STEP = Joi.alternatives().try(SCHEMA_STEP_STRING, SCHEMA_STEP_OBJECT);
 const SCHEMA_STEPS = Joi.array().items(SCHEMA_STEP).min(1);
-const SCHEMA_STEPS_NO_DUPS = Joi.array().items(SCHEMA_STEP).min(1).unique((a, b) => {
+// Template steps
+const SCHEMA_TEMPLATE_STEP_SUBOBJECT = SCHEMA_STEP_SUBOBJECT_BASE
+    .keys({
+        locked: Joi.boolean()
```

Comment:
should this be optional ? should be allow users to define the step object format without any attributes ?

Reply:
I think it's okay
the template-validator/config-parser can convert the non-attribute step objects to the simplified version